tests/rbd/rbd_trash_restore_image_having_clone.py

```diff
@@ -0,0 +1,176 @@
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def move_to_trash_and_restore(rbd, pool, image, sleep_time):
+    """
+    Move given image to trash, sleep for given time and restore image back from trash
+
+    Args:
+        rbd: RBD object
+        pool: pool in which image is present
+        image: image to be trashed
+        sleep_time: seconds to sleep before restoring
+    """
+    if rbd.move_image_trash(pool, image):
+        log.error(f"Moving image {image} to trash failed")
+        raise Exception(f"Moving image {image} to trash failed")
+
+    image_id = rbd.get_image_id(pool, image)
+
+    if not rbd.trash_exist(pool, image):
+        log.error("Image is not found in the Trash")
+        raise Exception("Image is not found in the Trash")
+    else:
+        log.info(f"Image {image} moved to trash")
+
+    sleep(sleep_time)
+
+    if rbd.trash_restore(pool, image_id):
+        log.error(f"Trash restore failed for image {image}")
+        raise Exception(f"Trash restore failed for image {image}")
+    return 0
+
+
+def rbd_trash_restore_image_having_clone(rbd, pool_type, **kw):
+    """
+    Verify parent image moved to trash and restored back when IO running on clone
+    and parent clone relationship still active.
+    Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: Key/value pairs of configuration information to be used in the test
+    """
+    pool = kw["config"][pool_type]["pool"]
+    image = kw["config"][pool_type]["image"]
+    snap = kw["config"][pool_type].get("snap", f"{image}_snap")
+    clone = kw["config"][pool_type].get("clone", f"{image}_clone")
+    try:
+        # Create snap and clone
+        rbd.snap_create(pool, image, snap)
+        snap_name_1 = f"{pool}/{image}@{snap}"
+        rbd.protect_snapshot(snap_name_1)
+        rbd.create_clone(snap_name_1, pool, clone)
+
+        log.info(
+            f"Checking parent-clone relationship for parent: {image}, clone: {clone}"
+        )
+        rbd_info = rbd.image_info(pool, clone)
+        if (
+            rbd_info.get("parent").get("image") == image
+            and rbd_info.get("parent").get("snapshot") == snap
+            and not rbd_info.get("parent").get("trash")
+        ):
+            log.info(
+                f"Parent-clone {image}-{clone} relationship is intact and parent is not in trash"
+            )
+        else:
+            log.error(f"Parent-clone {image}-{clone} relationship broken")
+            return 1
+
+        rbd_du = rbd.get_disk_usage_for_pool(pool, image=clone)
+        size_before_io = rbd_du["images"][0]["used_size"]
+        log.info(f"Clone size before IO : {size_before_io}")
+
+        client = kw["ceph_cluster"].get_nodes(role="client")[0]
+        log.info(
+            "Move parent image to trash when IO on Clone is in progress. (use rbd bench)"
```

Comment:
A small change, here we are referring "use rbd bench" but we actually using fio to run the IOs. This is confusing.

Reply:
Done. Removed that log use rbd bench.